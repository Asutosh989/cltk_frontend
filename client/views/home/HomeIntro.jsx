

HomeIntro = React.createClass({
	propTypes: {
		showSearchModal: React.PropTypes.func.isRequired,
	},

	render() {
		return (
			<section id="intro" className="cover fullscreen image-bg bg-dark ">

				<div className="background-image-holder less-blur blur">
					<img src="/images/column.jpg" role="presentation" />
				</div>
				<div className="background-screen light" />

				<div className="container v-align-transform header-container">
					<div className="row">

						<div className="header-center text-center">

							<h1>Explore open classical literature</h1>
							<input
								className="header-search"
								type="text"
								placeholder="Search . . ."
								readOnly
								onClick={this.props.showSearchModal}
								onTouchTap={this.props.showSearchModal}
								/>

							<h6 className="uppercase mb16">
								Read classical works in Greek, Latin, Chinese, Coptic, and Pali <br />
								and research metadata on your favorite texts.
							</h6>

							<a
								className="waves-effect waves-light btn-large"
								href="#beliefs" aria-label="Learn More"
							>
								Learn More
							</a>

						</div>

					</div>

				</div>

			</section>

		);
	},
});
