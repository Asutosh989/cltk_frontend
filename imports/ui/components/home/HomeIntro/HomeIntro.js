import React from 'react';

import PropTypes from 'prop-types';

function scrollDown() {
	$('html, body').animate({ scrollTop: $('#get-started').offset().top - 100 }, 300);
}

export default function HomeIntro(props) {
	return (
		<section id="intro" className="cover fullscreen image-bg bg-dark ">

			<div className="background-image-holder less-blur blur">
				<img src="/images/column.jpg" role="presentation" />
			</div>
			<div className="background-screen light" />

			<div className="container v-align-transform header-container">
				<div className="row">

					<div className="header-center text-center">

						<h1>Explore open classical literature</h1>
						<input
							className="header-search"
							type="text"
							placeholder="Search . . ."
							readOnly
							onClick={props.showSearchModal}
							onTouchTap={props.showSearchModal}
	/>

						<h6 className="uppercase mb16">
							Read and research works in Chinese, Coptic,
							Greek, Hebrew, Latin, Pali, Punjabi, Sanskrit,
							<br />
							Telugu, and Tibetan and contribute annotations to the community.
						</h6>

						<a
							className="waves-effect waves-light btn-large"
							href="#get-started"
							aria-label="Learn More"
							onClick={scrollDown}
						>
							Learn More
						</a>

					</div>

				</div>

			</div>

		</section>

	);
}

HomeIntro.propTypes = {
	showSearchModal: PropTypes.func.isRequired,
};
