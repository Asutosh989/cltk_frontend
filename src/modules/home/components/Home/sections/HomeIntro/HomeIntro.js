import React from 'react';
import PropTypes from 'prop-types';
import BackgroundImage from '../../../../../../components/common/cover/BackgroundImage';

export default function HomeIntro(props) {
	return (
		<section id="intro" className="cover fullscreen image-bg bg-dark ">

			<BackgroundImage
				imgSrc="/images/column.jpg"
			/>
			<div className="background-screen cyan" />

			<div className="container v-align-transform header-container">
				<div className="row">

					<div className="header-center text-center">

						<h1>Explore open classical literature</h1>
						<input
							className="header-search"
							type="text"
							placeholder="Search . . ."
							readOnly
							onClick={props.showSearchModal}
							onTouchTap={props.showSearchModal}
						/>
						<h6 className="uppercase mb16">
							Read and research works in Chinese, Coptic,
							Greek, Hebrew, Latin, Pali, Punjabi, Sanskrit,
							<br />
							Telugu, and Tibetan and contribute annotations to the community.
						</h6>
					</div>
				</div>
			</div>
		</section>

	);
}

HomeIntro.propTypes = {
	showSearchModal: PropTypes.func.isRequired,
};
