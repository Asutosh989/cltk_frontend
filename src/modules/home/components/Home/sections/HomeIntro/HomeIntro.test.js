
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import HomeIntro from './HomeIntro';

describe('HomeIntro', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<HomeIntro />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
