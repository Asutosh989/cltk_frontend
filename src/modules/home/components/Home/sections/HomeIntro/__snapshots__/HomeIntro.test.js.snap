// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeIntro renders correctly 1`] = `
<section
  className="cover fullscreen image-bg bg-dark "
  id="intro"
>
  <div
    className="background-image-holder less-blur blur"
  >
    <img
      role="presentation"
      src="/images/column.jpg"
    />
  </div>
  <div
    className="background-screen light"
  />
  <div
    className="container v-align-transform header-container"
  >
    <div
      className="row"
    >
      <div
        className="header-center text-center"
      >
        <h1>
          Explore open classical literature
        </h1>
        <input
          className="header-search"
          onClick={undefined}
          onTouchTap={undefined}
          placeholder="Search . . ."
          readOnly={true}
          type="text"
        />
        <h6
          className="uppercase mb16"
        >
          Read and research works in Chinese, Coptic, Greek, Hebrew, Latin, Pali, Punjabi, Sanskrit,
          <br />
          Telugu, and Tibetan and contribute annotations to the community.
        </h6>
        <a
          aria-label="Learn More"
          className="waves-effect waves-light btn-large"
          href="#get-started"
          onClick={[Function]}
        >
          Learn More
        </a>
      </div>
    </div>
  </div>
</section>
`;
